### ISSUE TITLE
micronaut-jms and micronaut-aws used different aws-sdk caused runtime error

### ISSUE THREAD

**Issue by yibo-long** [micronaut-projects/micronaut-platform#1133]
### Expected Behavior

We are using micronaut-gradle-plugin with importing micronaut platform 4.2.1. A microservice is using feature of [Bootstrap Configuration](https://docs.micronaut.io/latest/guide/#bootstrap) with aws secretmanager, and JMS with aws SQS.

We expect our gradle dependencies like this would work with micronaut bom sorting out all dependencies:

```
// build.gradle.kts
plugins {
    id("io.micronaut.minimal.application") version "4.2.1"
}

dependencies {
    ...
    // SQS dependencies:
    implementation("io.micronaut.jms:micronaut-jms-sqs")
    implementation("software.amazon.awssdk:sqs")

    // Bootstrap configuration dependencies:
    implementation("software.amazon.awssdk:sts")
    implementation("io.micronaut.aws:micronaut-aws-secretsmanager")
    implementation("io.micronaut.aws:micronaut-aws-sdk-v2")
    ...
}
```

### Actual Behaviour

All of our tests passed with the dependencies, however when deployed to the cloud where fetches the bootstrap configuration from aws secret manager, the service failed to start with the following log:

```
java.lang.NoClassDefFoundError: software/amazon/awssdk/protocols/query/interceptor/QueryParametersToBodyInterceptor
	at software.amazon.awssdk.services.sts.DefaultStsBaseClientBuilder.finalizeServiceConfiguration(DefaultStsBaseClientBuilder.java:73)
	at software.amazon.awssdk.awscore.client.builder.AwsDefaultClientBuilder.finalizeChildConfiguration(AwsDefaultClientBuilder.java:185)
	at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.syncClientConfiguration(SdkDefaultClientBuilder.java:193)
	at software.amazon.awssdk.services.sts.DefaultStsClientBuilder.buildClient(DefaultStsClientBuilder.java:39)
	at software.amazon.awssdk.services.sts.DefaultStsClientBuilder.buildClient(DefaultStsClientBuilder.java:28)
	at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.build(SdkDefaultClientBuilder.java:157)
	at software.amazon.awssdk.services.sts.internal.StsWebIdentityCredentialsProviderFactory$StsWebIdentityCredentialsProvider.<init>(StsWebIdentityCredentialsProviderFactory.java:74)
	at software.amazon.awssdk.services.sts.internal.StsWebIdentityCredentialsProviderFactory$StsWebIdentityCredentialsProvider.<init>(StsWebIdentityCredentialsProviderFactory.java:56)
	at software.amazon.awssdk.services.sts.internal.StsWebIdentityCredentialsProviderFactory.create(StsWebIdentityCredentialsProviderFactory.java:48)
	at software.amazon.awssdk.auth.credentials.WebIdentityTokenFileCredentialsProvider.<init>(WebIdentityTokenFileCredentialsProvider.java:115)
	at software.amazon.awssdk.auth.credentials.WebIdentityTokenFileCredentialsProvider.<init>(WebIdentityTokenFileCredentialsProvider.java:51)
	at software.amazon.awssdk.auth.credentials.WebIdentityTokenFileCredentialsProvider$BuilderImpl.build(WebIdentityTokenFileCredentialsProvider.java:315)
	at software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider.lambda$createChain$0(DefaultCredentialsProvider.java:97)
	at software.amazon.awssdk.utils.Lazy.getValue(Lazy.java:50)
	at software.amazon.awssdk.auth.credentials.internal.LazyAwsCredentialsProvider.resolveCredentials(LazyAwsCredentialsProvider.java:45)
	at software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider.resolveCredentials(DefaultCredentialsProvider.java:128)
	at software.amazon.awssdk.auth.credentials.AwsCredentialsProvider.resolveIdentity(AwsCredentialsProvider.java:54)
	at software.amazon.awssdk.identity.spi.IdentityProvider.resolveIdentity(IdentityProvider.java:60)
	at software.amazon.awssdk.auth.credentials.AwsCredentialsProviderChain.resolveCredentials(AwsCredentialsProviderChain.java:109)
	at software.amazon.awssdk.auth.credentials.AwsCredentialsProvider.resolveIdentity(AwsCredentialsProvider.java:54)
	at software.amazon.awssdk.identity.spi.IdentityProvider.resolveIdentity(IdentityProvider.java:60)
	at software.amazon.awssdk.awscore.internal.authcontext.AwsCredentialsAuthorizationStrategy.lambda$resolveCredentials$2(AwsCredentialsAuthorizationStrategy.java:112)
	at software.amazon.awssdk.core.internal.util.MetricUtils.measureDuration(MetricUtils.java:56)
	at software.amazon.awssdk.awscore.internal.authcontext.AwsCredentialsAuthorizationStrategy.resolveCredentials(AwsCredentialsAuthorizationStrategy.java:112)
	at software.amazon.awssdk.awscore.internal.authcontext.AwsCredentialsAuthorizationStrategy.addCredentialsToExecutionAttributes(AwsCredentialsAuthorizationStrategy.java:85)
	at software.amazon.awssdk.awscore.internal.AwsExecutionContextBuilder.invokeInterceptorsAndCreateExecutionContext(AwsExecutionContextBuilder.java:137)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.invokeInterceptorsAndCreateExecutionContext(AwsSyncClientHandler.java:67)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.lambda$execute$1(BaseSyncClientHandler.java:76)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.measureApiCallSuccess(BaseSyncClientHandler.java:182)
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:74)
	at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:45)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:53)
	at software.amazon.awssdk.services.secretsmanager.DefaultSecretsManagerClient.listSecrets(DefaultSecretsManagerClient.java:1159)
	at io.micronaut.aws.secretsmanager.SecretsManagerKeyValueFetcher.keyValuesByPrefix(SecretsManagerKeyValueFetcher.java:91)
	at io.micronaut.aws.distributedconfiguration.AwsDistributedConfigurationClient.getPropertySources(AwsDistributedConfigurationClient.java:83)
	at io.micronaut.discovery.config.DefaultCompositeConfigurationClient.lambda$getPropertySources$0(DefaultCompositeConfigurationClient.java:64)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at io.micronaut.discovery.config.DefaultCompositeConfigurationClient.getPropertySources(DefaultCompositeConfigurationClient.java:65)
	at io.micronaut.discovery.client.config.DistributedPropertySourceLocator.findPropertySources(DistributedPropertySourceLocator.java:75)
	at io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment.readPropertySourceList(DefaultApplicationContext.java:905)
	at io.micronaut.context.env.DefaultEnvironment.readPropertySources(DefaultEnvironment.java:439)
	at io.micronaut.context.env.DefaultEnvironment.start(DefaultEnvironment.java:277)
	at io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment.start(DefaultApplicationContext.java:891)
	at io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment.start(DefaultApplicationContext.java:860)
	at io.micronaut.context.DefaultApplicationContext.startEnvironment(DefaultApplicationContext.java:263)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:198)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:73)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:322)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:297)
	at io.span.services.pubsub.ApplicationKt.main(Application.kt:12)
Caused by: java.lang.ClassNotFoundException: software.amazon.awssdk.protocols.query.interceptor.QueryParametersToBodyInterceptor
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 56 common frames omitted
```

After a thorough debugging, we realized micronaut platform's BOM [4.2.1](https://github.com/micronaut-projects/micronaut-platform/blob/v4.2.1/gradle/libs.versions.toml#L15) used different awssdk version that has incompatible changes. micronaut-aws 4.1.1 is pointing to [2.20.162](https://github.com/micronaut-projects/micronaut-aws/blob/4.1.x/gradle/libs.versions.toml#L28), while micronaut-jms 3.1.0 is pointing to [2.21.15](https://github.com/micronaut-projects/micronaut-jms/blob/v3.1.0/gradle/libs.versions.toml#L12C20-L12C20)

We can force our `software.amazon.awssdk:sts` dependency to 2.21.15 for fix the dependency issue, but would be great to resolve it at the platform level.

### Steps To Reproduce

With the above gradle setup, run gradle :dependencies will show up:
```
|         +--- io.micronaut.aws:micronaut-aws-bom:4.1.1
|         |    +--- com.amazonaws:aws-java-sdk-bom:1.12.580
|         |    +--- software.amazon.awssdk:bom:2.20.162
|         |    |    +--- software.amazon.awssdk:sqs:2.20.162 -> 2.21.15 (c)
|         |    |    +--- software.amazon.awssdk:sts:2.20.162 (c)
```

### Environment Information

JDK Version: 21

### Version

4.2.1

**Comment by yibo-long** [comment]
I couldn't find the source code for awssdk in github, so only screenshots for the incompatible breaking change in DefaultStsBaseClientBuilder across awssdk version:

2.21.15
![image](https://github.com/micronaut-projects/micronaut-platform/assets/120519724/6dcb64c5-ac81-46b0-91e9-f47d4c20e809)


2.20.162
![image](https://github.com/micronaut-projects/micronaut-platform/assets/120519724/187ec98c-6b8f-4563-a713-feab107ce721)

**Comment by sdelamo** [comment]
Thanks @yibo-long I think the workaround of forcing the aws-sqs version is the only workaround for now.

**Comment by sdelamo** [comment]
thanks for reporting. This is fixed in Micronaut Framework 4.2.4.
